common/datavalues/src/data_array_scatter.rs

```diff
@@ -70,15 +71,67 @@ impl DataArrayScatter {
             )));
         }
 
-        let indices_array_values = Self::indices_values(indices)?;
-        Self::scatter_data(data, indices_array_values, nums)
+        match (data, indices) {
+            (DataColumnarValue::Array(data), DataColumnarValue::Array(indices)) => {
+                let indices_array_values = Self::indices_values(indices)?;
+                Self::scatter_data(data, indices_array_values, nums)
+            }
+            (DataColumnarValue::Constant(_, _), DataColumnarValue::Array(indices)) => {
```

Comment:
If data is constant, we just cut the rows size to be  `nums` ?

Reply:
They always have the same length.

For example:
```
data: Constant(1, 3),
indices: Array([1, 2, 1])
result: [Array([1, 1]), Array([1])]
```